From e184053fc9a3db726b8cedadd86360f67a6a679f Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@renesas.com>
Date: Tue, 7 Feb 2017 23:24:12 +0700
Subject: [PATCH 575/909] arm64: dts: r8a7795: Add AVS device node

commit 5ce7658132f6d6ee742fbd661809e69648c0606f from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

[dien.pham.ry: arm64: dts: r8a7795: Rename target_cpus to target_devices]

This patch supports AVS (Adaptive Voltage Switching)
for R8A7795 ES2.0 SoC.

To avoid checking for all cpu nodes,
that is redundant (e.g checking "a53_0,...")

This patch adds target_devices property to define clearly
what devices are proceeded by driver.

Signed-off-by: Van Do <van.do.xw@rvc.renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index 8bc01f490454..d1813d76eb99 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -781,6 +781,12 @@
 			reg = <0 0xe6060000 0 0x50c>;
 		};
 
+		avs: avs@e60a013c {
+			compatible = "renesas,r8a7795-avs", "renesas,rcar-gen3-avs";
+			reg = <0 0xe60a013c 0 0x04>;
+			target_devices = <&a57_0>, <&a57_1>, <&a57_2>, <&a57_3>;
+		};
+
 		cpg: clock-controller@e6150000 {
 			compatible = "renesas,r8a7795-cpg-mssr";
 			reg = <0 0xe6150000 0 0x1000>;
-- 
2.17.1

